[Desgin] PlaceListViewController UI

tea-hkim opened the pull request:
## 작업내용
<!-- 작업 내용과 이미지를 첨부해주세요. -->
- [x] header - segmentedControl을 커스텀하였습니다
- [x] 밑에는 TableView를 두었습니다
- [x] navigationBar를 커스텀하였습니다 

<!-- (+스크린샷)이 있다면 적어주세요. 없으면 지워주세요-->
|아이폰 13|아이폰 13 pro Max| 아이폰 se 3|
|:---:|:---:|:---:|
|<img width="250" src="https://user-images.githubusercontent.com/81206228/198209175-8b03cc00-f3ea-4e32-89c1-1b2f55d5a669.png">|<img width="250" src="https://user-images.githubusercontent.com/81206228/198209329-6adf07d6-20cc-4644-a954-688f20520aa7.png">|<img width="250" src="https://user-images.githubusercontent.com/81206228/198209268-1f0029a3-7272-4961-bb25-2b1b6ea9562f.png">|


## 리뷰포인트
<!-- 리뷰가 필요한 포인트와 해당 되는 커밋을 링크로 걸어주세요. -->
### SegmentedControl
- 현재 버튼에는 어떠한 기능도 구현되어 있지 않습니다 
- segmetedControl에도 addTarget의 역할을 하는 메서드가 있어서 '전체' 또는 '선정맛집'을 탭했을 때 탭이 된 segment Control의 인덱스를 받아와 로직을 구성할 수 있을 것 같습니다


## Checklist
- [x] 브랜치를 가져와 작업한 경우 이전 브랜치에 PR을 보냈는지 확인
- [x] 빌드를 위해 SceneDelegate 수정한 것 PR로 올리지 않았는지 확인
- [x] 필요없는 주석, 프린트문 제거했는지 확인
- [x] 컨벤션 지켰는지 확인
- [x] final, private 제대로 넣었는지 확인
- [x] 다양한 디바이스에 레이아웃이 대응되는지 확인
  - [x] iPhone SE
  - [x] iPhone 13
  - [x] iPhone 13 Pro Max

tea-hkim committed:
Feat:sparkles:: 커스텀 세그먼트컨트롤 구현

세그먼트컨트롤을 구현하였습니다
  - 세그먼트 컨트롤
  - 세그먼트 컨트롤 underLine

tea-hkim committed:
Feat:sparkles:: 물음표 이미지 추가

tea-hkim committed:
Feat:sparkles:: gray54 추가 및 물음표이미지 색상 변경

tea-hkim committed:
Chore:broom:: preview 디버그로 이동

tea-hkim committed:
Merge branch 'dev' of https://github.com/DeveloperAcademy-POSTECH/MacC-TEAM-ZESTY into feat/placeList/ZES-140-setting-segmeted-control

tea-hkim committed:
Feat:sparkles:: PlaceListViewController headerView 추가

tea-hkim committed:
Feat:sparkles:: Model Place 프로퍼티 변경

 - kakaoPlaceId 추가
 - creator nil에서 MockData로 변경

tea-hkim committed:
Feat:sparkles:: navigationBar 배경색 변경

Lia316 commented on Projects/App/Sources/UI/Place/PlaceList/View/PlaceListViewController.swift:
```diff
@@ -14,7 +14,16 @@ import SnapKit
 final class PlaceListViewController: UIViewController {
     
     // MARK: - Properties
+    
```
p4;

위에 투두는 수정 완료하셨으면 삭제 부탁드려요!

tea-hkim replied:
변경하였습니다! a5092c62df9fb3962e8c6d7451b9267831c1676c

Lia316 commented on Projects/App/Sources/UI/Place/PlaceList/View/PlaceListViewController.swift:
```diff
@@ -14,7 +14,16 @@ import SnapKit
+    
+    private let segmentIndicator = UIView()
+    private let segmentedControl = UISegmentedControl(items: ["전체", "선정맛집"])
+    private let questionMarkImage = UIImageView()
+    private let sideButtonImage = UIImageView()
```
p1;

...😱 버튼ㅇ..이 이미지??
버튼으로 고쳐주세욥‼️

tea-hkim replied:
변경하였습니다! 1edac500c92bc0984d102e1d946bd9e70f8d706e

Lia316 replied:
이 친구도 버튼 아닌가요?

tea-hkim replied:
변경했습니다! 0f3e9d0415cb707bae3e80a8a78ac56c5ee15629

Lia316 commented on Projects/App/Sources/UI/Place/PlaceList/View/PlaceListViewController.swift:
```diff
@@ -27,13 +36,82 @@ final class PlaceListViewController: UIViewController {
+            segmentIndicator.snp.remakeConstraints { (make) in
+                make.top.equalTo(segmentedControl.snp.bottom).offset(3)
+                make.height.equalTo(3)
+                make.width.equalTo(titlecount * 11)
+                make.centerX.equalTo(segmentedControl.snp.centerX).dividedBy(numberOfSegments / CGFloat(2.58 + CGFloat(selectedIndex-1.0)*2.0))
+            }
+        } else {
+            segmentIndicator.snp.makeConstraints { make in
+                make.top.equalTo(segmentedControl.snp.bottom).offset(3)
+                make.height.equalTo(3)
+                make.width.equalTo(20)
+                make.centerX.equalTo(segmentedControl.snp.centerX).dividedBy(segmentedControl.numberOfSegments)
```
이게...되네요? 👍🏻

Lia316 reviewed (changes requested):
세그 부드럽게 잘 움직이네요! 정말 예뻐요😆
수고 많으셨어요 닉!!

버튼만 수정 부탁드려요!

Lia316 commented on Projects/App/Sources/UI/Place/PlaceList/View/PlaceListViewController.swift:
```diff
@@ -42,10 +120,51 @@ extension PlaceListViewController {
+    
     private func createLayout() {
-        view.addSubview(tableView)
+        view.addSubviews([headerView, segmentedControl, segmentIndicator, questionMarkImage, tableView])
+        headerView.addSubviews([segmentedControl, segmentIndicator, questionMarkImage, sideButtonImage])
```
p3;

중복으로 addSubview 된 뷰들이 있어요..!

tea-hkim replied:
삭제하였습니다!
e8aa011bf0462cda277b852320b28d7b67e56ea7

Lia316 replied:
segmentIndicator와 questionMarkImage 는 여전히 중복 add되는 것 같아요~!

tea-hkim replied:
변경하였습니다! 0f3e9d0415cb707bae3e80a8a78ac56c5ee15629

Chaeho-Min commented on Projects/App/Sources/UI/Place/PlaceList/View/PlaceListViewController.swift:
```diff
@@ -42,10 +120,51 @@ extension PlaceListViewController {
+            make.height.equalTo(21)
+        }
+        
+        sideButtonImage.snp.makeConstraints { make in
+            make.verticalEdges.equalToSuperview().inset(10)
+            make.right.equalToSuperview()
+        }
```
p3;

sideButton이 고정된 크기가 있으니까 inset으로 제약조건을 주기보다는 너비와 높이를 직접 지정해주고 가운데 정렬하는게 더 좋을 것 같습니다!
괜히 짜부되는 일이 없도록!

```suggestion
        sideButtonImage.snp.makeConstraints { make in
            make.centerY.right.equalToSuperview()
            make.width.equalTo(70)
            make.height.equalTo(65)
        }
```

tea-hkim replied:
변경하였습니다 7281c930c08eb29f3a068d666e5166dd3e56c9bc

Chaeho-Min commented on Projects/App/Sources/UI/Place/PlaceList/View/PlaceListViewController.swift:
```diff
@@ -42,10 +120,51 @@ extension PlaceListViewController {
+            make.height.equalTo(85)
+        }
+        
+        segmentedControl.snp.makeConstraints { make in
+            make.verticalEdges.equalToSuperview().inset(27)
+            make.left.equalToSuperview().inset(20)
+            make.width.equalTo(172)
+        }
```
p5;

얘도 높이로? 근데 얘는 이미지같이 짜부되는 애는 아니라 뭐가 좋을지 잘 모르겠네요...

```suggestion
        segmentedControl.snp.makeConstraints { make in
            make.centerY.equalToSuperview()
            make.left.equalToSuperview().inset(20)
            make.width.equalTo(172)
            make.height.equalTo(32)
        }
```

tea-hkim replied:
7281c930c08eb29f3a068d666e5166dd3e56c9bc 변경하였습니다

Chaeho-Min commented on Projects/App/Sources/UI/Place/PlaceList/View/PlaceListViewController.swift:
```diff
@@ -42,10 +120,51 @@ extension PlaceListViewController {
+        
         tableView.snp.makeConstraints { make in
-            make.edges.equalToSuperview()
+            make.top.equalTo(headerView.snp.bottom).offset(20)
```
p1;

여기 offset 때문에 간격이 너무 넓네요 없애주세요!

```suggestion
            make.top.equalTo(headerView.snp.bottom)
```

tea-hkim replied:
7281c930c08eb29f3a068d666e5166dd3e56c9bc 변경하였습니다!

Chaeho-Min reviewed (changes requested):
이게 segmentedControl로 되네요...?
아주 사소한 변경점들 확인부탁요~

tea-hkim committed:
Feat:sparkles:: buttonImage를 button으로 수정

버튼 이미지를 버튼으로 변경하였습니다
 - 네비게이션 바에 있는 버튼들은 뷰가 완성되는 데로 추가하도록 하겠습니다
 - AddPlaceButton을 AddPlaceSearchViewController와 연결하였습니다

tea-hkim committed:
Merge branch 'dev' of https://github.com/DeveloperAcademy-POSTECH/MacC-TEAM-ZESTY into feat/placeList/ZES-140-setting-segmeted-control

tea-hkim committed:
Chore:broom:: navigationBarButton과 ProfileNickNameView 연결

 - configureNaviBar 연결

tea-hkim committed:
HOTFIX:ambulance:: 빌드 안되는 현상 해결

tea-hkim committed:
Chore:broom:: TODO 삭제

tea-hkim committed:
Chore:broom:: 중복된 addView 수정

tea-hkim committed:
Chore:broom:: layout 변경

 - segmentedControl의 높이를 지정해줬습니다
 - sideButtonImage의 높이를 지정하고 가운데 정렬을 해줬습니다
 - tableView offset 삭제

Chaeho-Min reviewed (approved):
LGTM

tea-hkim committed:
Feat:sparkles:: questionImage에서 questionButton으로 수정

Lia316 reviewed (approved):
꼭 수정 바라요 ㅠㅠㅠㅠㅠ 젭알!!!!!!!!!!!!!!!!
